tests/console/validate_selected_patterns.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validation module to check patterns installed.
+# Scenarios covered:
+# - zypper search for named patterns(base,enhanced_base;
+# - zypper search for removed pattern apparmor
+#
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+
+sub run {
+
+    select_console('root-console');
+
+    my @installed_patterns = split /,/, get_var('PATTERNS');
+    my %patterns           = (
+        base     => 'base',
+        minimal  => 'enhanced_base',
+        apparmor => 'apparmor',
+        x        => 'x11',
+        yast2    => 'yast2_basis'
+    );
+
+    foreach my $pattern (@installed_patterns) {
+        my $installed = ($pattern =~ s/^-//) ? 0 : 1;
+        if (!exists($patterns{$pattern})) {
+            die 'Pattern $pattern is not mapped: please add its exact name.';
+        }
+        elsif($installed) {
```

Comment:
You could use just `if` as previous condition will return from the run method. Matter of taste ;)

Reply:
in that case i just accepted the pull suggestion from @jknphy, thinking that it will make the review faster, instead of changing it.